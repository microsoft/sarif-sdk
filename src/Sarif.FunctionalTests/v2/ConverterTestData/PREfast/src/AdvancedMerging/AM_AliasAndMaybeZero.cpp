void MergeAliasAndZero ()
{
    int * source = 0;

    int * y = source;

    int result = *y;
}
