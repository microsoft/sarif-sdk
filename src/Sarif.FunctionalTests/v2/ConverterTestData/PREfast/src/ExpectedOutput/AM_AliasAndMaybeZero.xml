<?xml version='1.0' encoding='UTF-8'?>
<DEFECTS><DEFECT><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedmerging\</FILEPATH><FILENAME>am_aliasandmaybezero.cpp</FILENAME><LINE>7</LINE><COLUMN>8</COLUMN></SFA><DEFECTCODE>6011</DEFECTCODE><DESCRIPTION>Dereferencing NULL pointer 'y'. </DESCRIPTION><FUNCTION>MergeAliasAndZero</FUNCTION><DECORATED>?MergeAliasAndZero@@YAXXZ</DECORATED><FUNCLINE>1</FUNCLINE><PROBABILITY>1</PROBABILITY><RANK>4</RANK><CATEGORY><RULECATEGORY>mspft</RULECATEGORY></CATEGORY><PATH><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedmerging\</FILEPATH><FILENAME>am_aliasandmaybezero.cpp</FILENAME><LINE>3</LINE><COLUMN>10</COLUMN><KEYEVENT><ID>1</ID><KIND>declaration</KIND><IMPORTANCE>Essential</IMPORTANCE><MESSAGE>'source' is NULL</MESSAGE></KEYEVENT></SFA><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedmerging\</FILEPATH><FILENAME>am_aliasandmaybezero.cpp</FILENAME><LINE>5</LINE><COLUMN>10</COLUMN><KEYEVENT><ID>2</ID><KIND>alias</KIND><IMPORTANCE>Essential</IMPORTANCE><MESSAGE>(alias) 'y' receives the value of 'source' (which may be NULL)</MESSAGE></KEYEVENT></SFA><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedmerging\</FILEPATH><FILENAME>am_aliasandmaybezero.cpp</FILENAME><LINE>7</LINE><COLUMN>8</COLUMN><KEYEVENT><ID>3</ID><KIND>usage</KIND><IMPORTANCE>Essential</IMPORTANCE><MESSAGE>'y' is dereferenced, but may still be NULL</MESSAGE></KEYEVENT></SFA></PATH></DEFECT></DEFECTS>